lib/rspec/core/memoized_helpers.rb

```diff
@@ -153,6 +153,17 @@ def initialize
 
         def fetch_or_store(key)
           @memoized.fetch(key) do # only first access pays for synchronization
+            if @mutex.locked? && !@mutex.owned?
+              raise <<-EOS
```

Comment:
* The heredoc is making it difficult to see where this if statement ends (it looks like it might be wrapping the call to `synchronize`). Maybe try inlining? eg `raise <<EOS if @mutex.locked? && !@mutex.owned?`
* Is there an error class we should be using here?
* The dash on the heredoc isn't being used (looks like there's another heredoc in that file doing the same thing, IDK, don't really care, just pointing it out)